tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1119,42 +1119,59 @@ def _rostopic_list_bag(bag_file, topic=None):
                     break
 
 def _sub_rostopic_list(master, pubs, subs, publishers_only, subscribers_only, verbose, indent=''):
-    def topic_type(t, topic_types):
-        matches = [t_type for t_name, t_type in topic_types if t_name == t]
-        if matches:
-            return matches[0]
-        return 'unknown type'
-
     if verbose:
         topic_types = _master_get_topic_types(master)
 
         if not subscribers_only:
             print("\n%sPublished topics:"%indent)
-            for t, l in pubs:
+            for t, y, l in pubs:
                 if len(l) > 1:
-                    print(indent+" * %s [%s] %s publishers"%(t, topic_type(t, topic_types), len(l)))
+                    print(indent+" * %s [%s] %s publishers"%(t, y, len(l)))
                 else:
-                    print(indent+" * %s [%s] 1 publisher"%(t, topic_type(t, topic_types)))                    
+                    print(indent+" * %s [%s] 1 publisher"%(t, y))                    
 
         if not publishers_only:
             print(indent)
             print(indent+"Subscribed topics:")
-            for t,l in subs:
+            for t, y, l in subs:
                 if len(l) > 1:
-                    print(indent+" * %s [%s] %s subscribers"%(t, topic_type(t, topic_types), len(l)))
+                    print(indent+" * %s [%s] %s subscribers"%(t, y, len(l)))
                 else:
-                    print(indent+" * %s [%s] 1 subscriber"%(t, topic_type(t, topic_types)))
+                    print(indent+" * %s [%s] 1 subscriber"%(t, y))
         print('')
     else:
         if publishers_only:
-            topics = [t for t,_ in pubs]
+            topics = [t for t,_,_ in pubs]
         elif subscribers_only:
-            topics = [t for t,_ in subs]
+            topics = [t for t,_,_ in subs]
         else:
-            topics = list(set([t for t,_ in pubs] + [t for t,_ in subs]))                
+            topics = list(set([t for t,_,_ in pubs] + [t for t,_,_ in subs]))                
         topics.sort()
         print('\n'.join(["%s%s"%(indent, t) for t in topics]))
 
+def get_topic_list():
```

Comment:
It might be good to pass in the `master` since each caller already has a master variable.